rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -286,13 +292,32 @@ impl PendingOperation for PendingMessage {
                 self.metadata = Some(metadata_bytes.clone());
                 metadata_bytes
             }
-            Metadata::CouldNotFetch => {
-                return self.on_reprepare::<String>(None, ReprepareReason::CouldNotFetchMetadata);
-            }
-            // If the metadata building is refused, we still allow it to be retried later.
-            Metadata::Refused(reason) => {
-                warn!(?reason, "Metadata building refused");
-                return self.on_reprepare::<String>(None, ReprepareReason::MessageMetadataRefused);
+            Metadata::Failed(err) => {
+                match err {
+                    MetadataBuildError::CouldNotFetch => {
+                        return self
+                            .on_reprepare::<String>(None, ReprepareReason::CouldNotFetchMetadata);
+                    }
+                    // If the metadata building is refused, we still allow it to be retried later.
+                    MetadataBuildError::Refused(reason) => {
+                        warn!(?reason, "Metadata building refused");
+                        return self
+                            .on_reprepare::<String>(None, ReprepareReason::MessageMetadataRefused);
+                    }
+                    // These errors cannot be recovered from, so we drop them
+                    MetadataBuildError::UnsupportedModuleType(reason) => {
+                        warn!(?reason, "Unsupported module type");
+                        return PendingOperationResult::Drop;
+                    }
+                    MetadataBuildError::MaxDepthExceeded(depth) => {
+                        warn!(depth, "Max ISM depth reached");
+                        return PendingOperationResult::Drop;
+                    }
```

Comment:
Previously, we called `on_reprepare` to deal with these two errors, but now we drop the operation straight away. Is this new behaviour expected/intentional?

Reply:
Yeah thats a good point. What do you think @daniel-savu @tkporter ?
Intuitively, it made sense to me to drop these if they are unprocessable with the ISM limits we have.

Reply:
imo we should not drop these and just reprepare, a recipient can always change the ISM configuration and we should allow for that to happen